### ISSUE TITLE
Memory leak: UICollectionView

### ISSUE THREAD

**Issue by Sebastian-Hojas** [ReactiveCocoa/ReactiveCocoa#3356]
Hi!

When I was trying to setup a CollectionView with 5.0.0-alpha.5 I faced the following problem: With a large list of items, I leak massively signals and, unless I am misusing the framework, I think comes from the current alpha: All I do is set a simple producer to a binding target with a lifetime of `prepareForReuse`. I created a simple project to isolate the problem. Here's the code: 

```
let property = MutableProperty<UIColor>(UIColor.red)
override public func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell
{
    let cell = collectionView.dequeueReusableCell(withReuseIdentifier: "reuse", for: indexPath)
        
    cell.reactive.backgroundColor <~
        property.producer.take(until: cell.reactive.prepareForReuse)   
    return cell
}
```

This is enough to reproduce the leak of Signals, AliveState, ActionDisposable. I can provide the instrument trace or an example project if necessary. 
Interestingly, if I keep track of the disposables myself with a custom cell and a `Disposable`, it seems _better_: 

```
class BindableCell: UICollectionViewCell
{
    var disposable: Disposable?
    
    required init?(coder aDecoder: NSCoder) {
        super.init(coder: aDecoder)
    }
    
    override func prepareForReuse() {
        disposable?.dispose()
        disposable = nil
    }
    
    func bind(producer: SignalProducer<UIColor, NoError>)
    {
        disposable = self.reactive.backgroundColor <~ producer
    }
   
}
```
This still leaks, but I have the impression that fewer signals leak. 

Do I misuse the framework? Would anything help you to trace down the problem better?

**Comment by andersio** [comment]
Xcode Instruments would report false positives as it does not handle Swift's weak retaining correctly. Unless you see a memory growth when you repeatedly scroll through your collection views, it would be a legitimate one.

**Comment by Sebastian-Hojas** [comment]
For the first version with `.take(until:)` the memory demand increases significantly for every `collectionView(collectionView:cellForItemAt:)` (not just for the first call when they get initiated).
For the second attempt with a `Disposable` it does increase, but only marginaly (though I still assume that it leaks).

**Comment by andersio** [comment]
Thanks for the report. I have reproduced it. Investigating.

**Comment by andersio** [comment]
Hmm, I think I have found the culprit. Will patch it ASAP.

**Comment by andersio** [comment]
We have patched it in ReactiveSwift. A new release would come ASAP.

**Comment by andersio** [comment]
Fixed in 5.0.0-alpha.6.

**Comment by Abhishek12345679** [comment]
????
